ycms/cms/views/ward/discharge_view.py

```diff
@@ -0,0 +1,29 @@
+import datetime
+
+from django.http import JsonResponse
+from django.views.generic import View
+
+from ...models import Bed, BedAssignment
+
+
+class DischargePatientView(View):
+    """
+    View to discharge a patient
+    """
+
+    def post(self, request, *args, **kwargs):
+        """
+        This function discharges a patient
+        """
+        # discharged = request.POST.get('discharged')
+        bed = Bed.objects.get(pk=kwargs["bed_id"])
+        bed_assignment = BedAssignment.objects.get(pk=kwargs["assignment_id"])
+
+        bed.is_available = True
+        bed_assignment.discharge_date = datetime.datetime.now()
+
+        bed_assignment.save()
+        bed.save()
+
+        return JsonResponse({"success": True})
+        # return redirect(reverse('ward_detail', args=[kwargs['pk']]))
```

Comment:
Since the availability is now implicit:

```suggestion
import datetime

from django.http import JsonResponse
from django.views.generic import View

from ...models import BedAssignment


class DischargePatientView(View):
    """
    View to discharge a patient
    """

    def post(self, request, *args, **kwargs):
        """
        This function discharges a patient
        """
        bed_assignment = BedAssignment.objects.get(pk=kwargs["assignment_id"])
        bed_assignment.discharge_date = datetime.datetime.now()
        bed_assignment.save()

        return JsonResponse({"success": True})
```